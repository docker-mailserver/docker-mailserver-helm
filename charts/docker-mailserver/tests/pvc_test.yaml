suite: pvc creation
templates:
  - pvc.yaml
values:
  - ../values.yaml
chart:
  version: 0.1.0
  appVersion: 0.1.0
tests:
  - it: should apply annotations from persistent_volume_claims.annotations
    set:
      persistent_volume_claims:
        mail-config:
          annotations:
            backup.banana.io/deltas: pancakes
    asserts:
      - matchSnapshot: { }

  - it: should create pvc of specified size
    set:
      persistent_volume_claims:
        mail-data:
          size: 1Pb
    documentIndex: 1
    asserts:
      - equal:
          path: spec.resources.requests.storage
          value: 1Pb

  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: { }