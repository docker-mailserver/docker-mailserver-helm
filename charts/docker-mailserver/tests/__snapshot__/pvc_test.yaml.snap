manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-config
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Mi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-data
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
  3: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-log
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
  4: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-state
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
should apply annotations from persistent_volume_claims.annotations:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations:
        backup.banana.io/deltas: pancakes
      name: RELEASE-NAME-docker-mailserver-mail-config
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Mi
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-data
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
  3: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-log
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
  4: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      name: RELEASE-NAME-docker-mailserver-mail-state
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
